File: weap-gestion-documental-backend/src/main/java/com/weap/gd/entity/CarguePlano.java
package com.weap.gd.entity;

import java.time.LocalDate;
import java.time.LocalDateTime;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import lombok.Data;

@Data
@Entity
@Table(name = "cargue_plano", schema = "cargue")
public class CarguePlano {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	
	private String nombreArchivo;
	
	@Column(name = "nit_prestador")
	private String nitPrestador;
	
	private LocalDateTime fechaCargue;
	
	private Boolean erroresEnCargue;
	
	@Column(name = "numero_registro")
	private Integer numeroRegistro;
	
	@JsonIgnoreProperties
	@OneToOne(cascade = CascadeType.ALL)
	@JoinColumn(name = "job_execution_id")
	private EjecucionTarea ejecucionTarea;	
	
	private String usuario;
	
	@Column(name = "delete_at")
	private LocalDate deleteAt;

}

File: weap-gestion-documental-frontend/src/app/models/cargue/plano/cargue-plano.ts
import { EjecucionTarea } from "../../ejecucion-tarea";

export class CarguePlano {
    id: number;
    nombreArchivo?: string;
    nitPrestador?: string;
    codigoPrestador!: string;
    fechaCargue?: Date;
    erroresEnCargue?: boolean;
    ejecucionTarea?: EjecucionTarea;
    numeroRegistro?: String;
    usuario?: string;
    fechaBaja?: string;
}
